src/System.Runtime.Analyzers/Core/SpecifyCultureInfo.cs

```diff
@@ -24,15 +27,86 @@ public abstract class SpecifyCultureInfoAnalyzer : DiagnosticAnalyzer
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterCompilationStartAction(csaContext =>
+            {
+                var cultureInfoType = csaContext.Compilation.GetTypeByMetadataName("System.Globalization.CultureInfo");
+                if (cultureInfoType != null)
+                {
+                    csaContext.RegisterOperationAction(oaContext =>
+                    {
+                        var invocationExpression = (IInvocationExpression)oaContext.Operation;
+                        var targetMethod = invocationExpression.TargetMethod;
+                        if (targetMethod.ContainingType == null)
+                        {
+                            return;
+                        }
+
+                        var methodsWithSameNameAsTargetMethod = targetMethod.ContainingType.GetMembers(targetMethod.Name).OfType<IMethodSymbol>();
+                        if (methodsWithSameNameAsTargetMethod.Count() < 2)
+                        {
+                            return;
+                        }
+
+                        var correctOverloads = methodsWithSameNameAsTargetMethod.Where(candidateMethod =>
+                        {
+                            if (targetMethod.Parameters.Count() + 1 != candidateMethod.Parameters.Count())
+                            {
+                                return false;
+                            }
+
+                            // Assume CultureInfo is the last parameter so j, which is the index of the parameter
+                            // in m to compare against targetMethod's parameter is set to 0
+                            int j = 0;
+
+                            if (candidateMethod.Parameters.First().Type.Equals(cultureInfoType))
+                            {
+                                // If CultureInfo is the first then the parameters to compare in m against targetMethod
+                                // is offset by 1
+                                j = 1;
+                            }
+                            else if (!candidateMethod.Parameters.Last().Type.Equals(cultureInfoType))
+                            {
+                                return false;
+                            }
+
+
+                            for (int i = 0; i < targetMethod.Parameters.Count(); i++, j++)
+                            {
+                                if (!targetMethod.Parameters[i].Type.Equals(candidateMethod.Parameters[j].Type))
+                                {
+                                    return false;
+                                }
+                            }
+
+                            return true;
+                        });
+
+                        // Pick the one with CultureInfo as the last Parameter
```

Comment:
Clarify comment: "`If there are two matching overloads, one with CultureInfo as the first parameter and one with CultureInfo as the last parameter, arbitrarily report the diagnostic on the overload with CultureInfo as the first parameter." (At least, I assume the choice is arbitrary.)